### ISSUE TITLE
Usage with sampling or data augmentation techniques.

### ISSUE THREAD

**Issue by ydennisy** [cgnorthcutt/cleanlab#61]
Hello and thank you for a great package!

I was wondering if you have any advice or best practice surrounding how best to use this in harmony with something like SMOTE or a data augmentation step.

Intuitively it seems the best approach would be to first filter out any noisy examples - then do any oversampling or augmentation.

Do you have any further best practices into the sort of flow? I have a text classification problem, with hugely imbalanced classes.

Thanks in advance!

**Comment by cgnorthcutt** [comment]
Hi @ydennisy , your approach sounds reasonable -- first clean (prune out noisy data), then augment the cleaned dataset.

I think you can do something smarter, though. Most data augmentation is in the data space (not the label space). 

Nowadayas, the best supervised learning often fine-tunes a model that has already been pre-trained in an unsupervised manner (auto-encoders, self-supervision, predicting the next item in a sequence, etc), using no labels. Using this approach, you can train on all the data (augmented), ignoring labels. Then once you have a pre-trained model, you can clean out the noise, and fine-tune on your clean labeled data.

I've done this and cleanlab works quite well for it. But your task needs to take advantage of unsupervised or self-supevised pre-training.

**Comment by ydennisy** [comment]
Thanks @cgnorthcutt 

Yes this does seem to be the approach most often seen, yet for some reason I am yet to have any great improvements fine-tuning something like BERT vs doc2vec followed by a LogReg model on top.

However converting the whole problem to a self-supervised problem is the dream :)
